cli/tests/upgrade.bats

```diff
@@ -232,3 +232,21 @@ EOF
   assert_success
   assert_output "⬆️  Upgraded 'hello' in environment 'test'."
 }
+
+# bats test_tags=upgrade:flake:iid
+@test "upgrade for flake installable by iid" {
+  "$FLOX_BIN" init
+
+  MANIFEST_CONTENTS="$(cat << "EOF"
+  version = 1
+
+  [install]
+  hello.flake = "github:NixOS/nixpkgs#hello"
+EOF
+  )"
+
+  echo "$MANIFEST_CONTENTS" | "$FLOX_BIN" edit -f -
+
+  run "$FLOX_BIN" upgrade hello
+  assert_success
```

Comment:
This would have saved me running it locally to verify that we got the expected behaviour and not just the absence of the error. If we're pinning then we also know that this won't be upgraded if nixpkgs advances during the test:
```suggestion
  assert_success
  assert_output "ℹ️  The specified packages do not need to be upgraded in environment 'test'."
```